pandas/core/reshape/concat.py

```diff
@@ -717,8 +717,8 @@ def _make_concat_multiindex(indexes, keys, levels=None, names=None) -> MultiInde
         new_levels.extend(new_index.levels)
         new_codes.extend([np.tile(lab, kpieces) for lab in new_index.codes])
     else:
-        new_levels.append(new_index)
-        new_codes.append(np.tile(np.arange(n), kpieces))
+        new_levels.append(new_index.unique())
+        new_codes.append(np.tile(new_index.unique().get_indexer(new_index), kpieces))
```

Comment:
nitpick: can we split this into multiple lines?

Reply:
Sure, like this=

Reply:
yah, thanks